python/ray/data/_internal/planner/map_batches.py

```diff
@@ -18,7 +18,7 @@ def generate_map_batches_fn(
     batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
     prefetch_batches: int = 0,
     zero_copy_batch: bool = False,
-) -> Callable[[Iterator[Block]], Iterator[Block]]:
+) -> Callable[[Iterator[Block], BatchUDF], Iterator[Block]]:
```

Comment:
@c21 This UDF is technically `def transform(blocks: Iterator[Block], fn: BatchUDF, *args, **kwargs) -> Iterator[Block])`, thinking about updating this and others with a `typing.Protocol` to capture this.

Reply:
Yeah I noticed it today, let's do a followup PR to fix these types. Seems to be minor.